src/HatsEligibilitiesChain.sol

```diff
@@ -134,13 +135,9 @@ contract HatsEligibilitiesChain is HatsEligibilityModule {
           eligibleInClause = false;
         }
 
-        unchecked {
-          ++moduleIndex;
-        }
+        moduleOffset += 20; // increment to the next clause
```

Comment:
can make this unchecked since the gas limit will be reached long before it ever overflows

Reply:
<img width="594" alt="Screenshot 2023-08-08 at 21 50 05" src="https://github.com/Hats-Protocol/hats-module/assets/81111572/b2580b6d-574e-47cb-8795-15f4d15e965a">

`moduleOffset` is calculated using `NUM_CONJUCTION_CLAUSES` which is given as a user input, and so can theoretically be close to the `uint256.max` if someone tries to mess up with it